warp-contracts-plugin-quickjs/src/eval/QuickJsEvaluator.ts

```diff
@@ -50,6 +50,42 @@ export class QuickJsEvaluator {
     randomHandle.dispose();
   }
 
+  dummyPromiseEval() {
+    const dummyPromiseEval = this.vm.newFunction('dummyPromise', () => {
+      const promise = this.vm.newPromise();
+      promise.resolve(this.vm.newString(''));
+      promise.settled.then(this.vm.runtime.executePendingJobs);
+      return promise.handle;
+    });
+    this.vm.setProp(this.vm.global, 'dummyPromise', dummyPromiseEval);
+    dummyPromiseEval.dispose();
+  }
+
+  evalExternal() {
+    const readExternalHandle = this.vm.newFunction('readExternal', (processIdHandle, actionHandle) => {
+      const promise = this.vm.newPromise();
+      this.readExternal(processIdHandle, actionHandle).then((result) => {
+        promise.resolve(this.vm.newString(JSON.stringify(result) || ''));
+      });
+      promise.settled.then(this.vm.runtime.executePendingJobs);
+      return promise.handle;
+    });
+    this.vm.setProp(this.vm.global, 'readExternal', readExternalHandle);
+    readExternalHandle.dispose();
+  }
+
+  async readExternal(processIdHandle: QuickJSHandle, actionHandle: QuickJSHandle) {
+    const processId = this.vm.getString(processIdHandle);
+    const action = this.vm.getString(actionHandle);
+    const { dryrun } = await import('@permaweb/aoconnect');
+    const readRes = await dryrun({
```

Comment:
what is the default timeout for the dryrun? maybe we should it limit to sth. like 10s?

Reply:
i dont know what's the default but i've added some logic so the error is thrown if dryrun is executing more than 10s